File: tests/fixtures/real_world/java/state/double_close.java
import java.io.*;

public class DoubleClose {
    public void doubleCloseStream(String path) throws IOException {
        FileInputStream fis = new FileInputStream(path);
        fis.close();
        fis.close();
    }

    public void useAfterClose(String path) throws IOException {
        FileInputStream fis = new FileInputStream(path);
        fis.close();
        byte[] data = new byte[1024];
        fis.read(data);
    }
}

File: tests/fixtures/real_world/go/state/double_close.go
package main

import "os"

func doubleClose(path string) {
	f, _ := os.Open(path)
	f.Close()
	f.Close()
}

func useAfterClose(path string) {
	f, _ := os.Open(path)
	f.Close()
	buf := make([]byte, 1024)
	f.Read(buf)
}
